### PR TITLE
Arrays

### PR DESCRIPTION
**Pull request by ericlesaquiles**
Mostly just commented the tests that would crash euslisp and listed a few of the issues (on the README.org).

### CODE REVIEW

File: arrays/adjust-array.lsp [hunk 1]
```diff
@@ -567,33 +567,33 @@
...
+;;          (a1 (make-array nil :displaced-to a0 :element-type 'character))
+;;          (a2 (adjust-array a1 nil :element-type 'character)))
+;;     a2)
+;;    #.(make-array nil :initial-element #\x :element-type 'character))
 
-(def-adjust-array-test adjust-array.string.adjustable.1
-  (5 :initial-contents "abcde" :adjustable t :element-type 'character)
-  (4 :element-type 'character)
-  "abcd")
+(def-adjust-array-te;; st adjust-array.string.adjustable.1
```
Comment (furushchev) [inline arrays/adjust-array.lsp:596]: Is this just a typo?

Comment (ericlesaquiles) [reply arrays/adjust-array.lsp:596]: jesus.. how could i have overlooked this

**Comment by Affonso-Gui** [comment]
It seems that we have too much `#0a` forms in the array tests, maybe for testing the class structure in a fast way.

Instead of commenting out all of the code that use it, what if we try to:
- Implement `#0a` at eus
or
- Change the tests to `#1a`

If we opt for implementing it on eus, we can make something like [this](https://github.com/Affonso-Gui/EusLisp/commit/b3baaa844729baf0c54cb60a24338d370cda9b05). However, not sure if it is a good idea. @furushchev Any thoughts?

**Comment by Affonso-Gui** [comment]
Maybe it is more interesting to have `errors` signing what went wrong on each function, instead of commenting it all out. This way we can know that there is a function to be tested, but something is not going right.

Refer to: https://github.com/euslisp/ansi-test/blob/gsoc2018/data-and-control-flow/and.lsp#L8-L11

**Comment by ericlesaquiles** [comment]
I'm really sorry for the long delay. 

About the #0A, overall I think it might not be a very good idea to make a change on the tests so as to make them pass (though making some signaling to show they're failing while stopping them from breaking the tests does seem like a good idea), unless there is no alternative. 

Implementing seem to me a better idea. The one you did, @Affonso-Gui , seems to me to work (they did work on the few tests I performed, and I don't see side-effects, but it could be just me..). Maybe you'd like to make a PR? I guess it might facilitate eventual discussions about it. 

By the way, wondering about the usage of zero dimensional vectors, I got to know that in Julia people use them as a means to pass values by reference. I guess some similar behavior might be expected in Common Lisp.

**Review by Affonso-Gui — APPROVED** [review]

**Pull request merged**